### PR TITLE
Support running container as non-root

### PR DESCRIPTION
**Pull request by onitake**
This PR allows running the container as non-root, to support restrictive PSPs.

* Create a non-root runtime user
* Specify execution paths explicitly
* Install only files needed by the application into the container

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
+RUN useradd -d /app -m appuser
+USER appuser
+
 WORKDIR /app
+
 COPY Pipfile* /app/
-RUN cd /app && \
-    pipenv install --system --deploy
+COPY *.py /app/
+COPY LICENSE /app/
```
Comment (joschi36) [inline Dockerfile]:
```suggestion
COPY *.py LICENSE /app/
```

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,19 @@
 FROM python:3.8-slim
-RUN pip install --no-cache-dir pipenv
+
+RUN useradd -d /app -m appuser
+USER appuser
```
Comment (joschi36) [inline Dockerfile]: Kubernetes with a pod security policy runAsNonRoot unfortunatly can't validate if user is non-root if it isn't numeric.
distroless for instance sets 65532. https://github.com/GoogleContainerTools/distroless/blob/main/base/base.bzl

You would need to specify runAsUser so it works.

Comment (joschi36) [reply Dockerfile]: As an alternative bitnami's python image most likely will have non-root preconfigured.

**Review by joschi36 — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (onitake) [reply Dockerfile]: @joschi36 Do you want me to do the same COPY consolidation for all the other non-root PRs?

File: Dockerfile (hunk 2 above)
Comment (onitake) [reply Dockerfile]: I'm not too fond of their image, and it also doesn't cover the other containers that don't need a Python environment, such as hbbtv_stream_check_api.
Also, I don't quite understand what the issue is. Do you mean that the "USER appuser" statement will stay as-is in the resulting container manifest? I can hard-code a numeric uid if this helps.

Comment (onitake) [reply Dockerfile]: And to justify _why_ I don't like the bitnami images: The way they're building their minideb base image is very obscure and hackish. I'd prefer that we don't use something that is hard to understand. I also don't see how they implemented their rootless setup - it's certainly not done in the Dockerfile: https://github.com/bitnami/bitnami-docker-python/blob/3.9.5-prod-debian-10-r47/3.9-prod/debian-10/Dockerfile

Comment (joschi36) [reply Dockerfile]: It just needs to be numeric

Comment (joschi36) [reply Dockerfile]: https://github.com/kubernetes/kubernetes/issues/59819

Comment (onitake) [reply Dockerfile]: The USER flag is now numeric (uid 65532).

**Comment by onitake** [comment]
@joschi36 Resolved the conflict introduced by the Py3.9 upgrade.

Can you review the changes, please?

File: Dockerfile [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
+
+RUN useradd -u 65532 -d /app -m appuser
+USER 65532
+
 WORKDIR /app
+
 COPY Pipfile* /app/
-RUN cd /app && \
-    pipenv install --system --deploy
+COPY *.py LICENSE /app/
```
Comment (joschi36) [inline Dockerfile]: I think we can combine these copies

**Review by joschi36 — COMMENTED** [review]
Minor comment on Dockerfile

File: Dockerfile (hunk 3 above)
Comment (onitake) [reply Dockerfile]: I considered it at first... but should we really do it in the same PR?
Remember, this one is for running as non-root.
I'm ok with changing it here though.

Comment (joschi36) [reply Dockerfile]: I think combining these are ok. :)

**Review by joschi36 — APPROVED** [review]

**Pull request merged**